src/utils/extractWikiLinks.spec.ts

```diff
@@ -8,71 +8,200 @@ import { extractWikiLinks } from "./extractWikiLinks";
 // TODO test with other remark plugins e.g. original wiki links
 
 describe("extractWikiLinks", () => {
+  const from = "abc/foobar.md";
+
   describe("Common Mark links", () => {
     test("should extract CommonMark links", () => {
-      const source = "[Page 1](page-1) [Page 2](page-2) [Page 3](page-3)";
+      const source = "[Page 1](page-1)";
+      const links = extractWikiLinks(from, source);
       const expectedLinks = [
-        { linkType: "normal", linkSrc: "page-1" },
-        { linkType: "normal", linkSrc: "page-2" },
-        { linkType: "normal", linkSrc: "page-3" },
+        {
+          from: "abc/foobar.md",
+          to: "abc/page-1",
+          toRaw: "page-1",
+          text: "Page 1",
+          embed: true,
+          internal: true,
+        },
       ];
-      const links = extractWikiLinks(source);
-      expect(links).toHaveLength(expectedLinks.length);
-      links.forEach((link) => {
-        expect(expectedLinks).toContainEqual(link);
-      });
+      expect(links).toEqual(expectedLinks);
     });
 
-    test("should extract embed type CommonMark links", () => {
-      const source = "![abc](My_File.png)";
-      const expectedLinks = [{ linkType: "embed", linkSrc: "My_File.png" }];
-      const links = extractWikiLinks(source);
-      expect(links[0]).toEqual(expectedLinks[0]);
+    test("should extract CommonMark links with image extension", () => {
+      const source = "[hello](world.png)";
+      const links = extractWikiLinks(from, source);
+      const expectedLinks = [
+        {
+          from: "abc/foobar.md",
+          to: "abc/world.png",
+          toRaw: "world.png",
+          text: "hello",
+          embed: true,
+          internal: true,
+        },
+      ];
+      expect(links).toEqual(expectedLinks);
```

Comment:
`[hello](world.png)` is not an embed format, it's a regular CommonMark link. It should have a leading `!` for it to be an embed. So, if you want to test if it gets parsed as a regular link:

```suggestion
    test("should extract CommonMark links with image extension", () => {
      const source = "[hello](world.png)";
      const links = extractWikiLinks(from, source);
      const expectedLinks = [
        {
          from: "abc/foobar.md",
          to: "abc/world.png",
          toRaw: "world.png",
          text: "hello",
          embed: false,
          internal: true,
        },
      ];
      expect(links).toEqual(expectedLinks);
```